src/H5.c

```diff
@@ -896,17 +896,17 @@ H5_check_version(unsigned majnum, unsigned minnum, unsigned relnum)
                 break;
         } /* end switch */
 
-    } /* end if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum) */
+    } /* end if (H5_VERS_MAJOR != majnum) */
 
-    /* H5_VERS_RELEASE should be compatible, we will only add checks for exceptions */
-    /* Library develop release versions are incompatible by design */
-    if (H5_VERS_RELEASE != relnum) {
-        for (unsigned i = 0; i < VERS_RELEASE_EXCEPTIONS_SIZE; i++) {
+    /* H5_VERS_MINOR should be compatible, we will only add checks for exceptions */
+    /* Library develop minor versions are incompatible by design */
```

Comment:
For semantic versioning, minor version increments should be backward compatible, so I'm not sure why the PR treats minor version differences as exceptions that need special handling

Reply:
Right, we want to update the schema for semantic versioning,

As is the PR, it does not distinguish between:
 2.0.x -> 2.1.x  is ok
2.1.x -> 2.-.x should warn/fail

it should be 
(1) major must match
(2) minor version backward compatibility check
     2a, check if either version is in the incompatible list (e.g., develop)